core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -361,6 +345,14 @@ public void setMessageCompression(boolean enabled) {
   }
 
   @Override
+  public void setCompression(String compressorName) {
+    checkState(compressor == null, "compressor already set");
```

Comment:
to avoid it being called erroneously after network traffic has begun.  Setting it twice seems like an error anyways.